mathesar_ui/src/systems/table-view/table-inspector/column/SetDefaultValue.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+  import DynamicInput from '@mathesar/components/cell-fabric/DynamicInput.svelte';
+  import EditableTextWithActions from '@mathesar/components/EditableTextWithActions.svelte';
+  import {
+    getTabularDataStoreFromContext,
+    type ProcessedColumn,
+  } from '@mathesar/stores/table-data';
+  import { toast } from '@mathesar/stores/toast';
+
+  export let column: ProcessedColumn;
+
+  const tabularData = getTabularDataStoreFromContext();
+  $: ({ columnsDataStore } = $tabularData);
+
+  async function save(value: unknown) {
+    try {
+      await columnsDataStore.patch(column.id, {
+        default: {
+          is_dynamic: !!column.column.default?.is_dynamic,
+          value,
+        },
+      });
+    } catch (err) {
+      const message =
+        err instanceof Error
+          ? err.message
+          : 'Unable to change column display options.';
+      toast.error(message);
+    }
+  }
+</script>
+
+<div class="default-value-container">
+  <span class="label">Value</span>
+  <EditableTextWithActions
+    initialValue={column.column.default?.value}
+    onSubmit={save}
+    getValidationErrors={() => []}
+    inputComponentAndProps={{
+      component: DynamicInput,
+      props: {
+        componentAndProps: column.inputComponentAndProps,
+      },
+    }}
+  />
```

Comment:
I'd like the `EditableTextWithActions` component to not be used here (or modified) to support any input for both display and editing purpose, and preferably using slots instead of props.